invisible_cities/io/mcinfo_io.py

```diff
@@ -0,0 +1,311 @@
+
+            break
+
+    return hit_rows, particle_rows
+
+
+def read_mcinfo(h5f, event_range=(0, int(1e9))) ->Mapping[int, Mapping[int, MCParticle]]:
+    h5extents   = h5f.root.MC.extents
+    h5hits      = h5f.root.MC.hits
+    h5particles = h5f.root.MC.particles
+
+    mc_info = (h5extents, h5hits, h5particles)
+    events_in_file = len(h5extents)
+
+    all_events = {}
+
+    for iext in range(*event_range):
+        if iext >= events_in_file:
+            break
+
+        current_event = {}
+        evt_number = h5extents[iext]['evt_number']
+        hit_rows, particle_rows = read_mcinfo_evt(mc_info, evt_number, iext)
+
+        for h5particle in particle_rows:
+            this_particle = h5particle['particle_indx']
+            current_event[this_particle] = MCParticle(h5particle['particle_name'].decode('utf-8','ignore'),
+                                                      h5particle['primary'],
+                                                      h5particle['mother_indx'],
+                                                      h5particle['initial_vertex'],
+                                                      h5particle['final_vertex'],
+                                                      h5particle['initial_volume'].decode('utf-8','ignore'),
+                                                      h5particle['final_volume'].decode('utf-8','ignore'),
+                                                      h5particle['momentum'],
+                                                      h5particle['kin_energy'],
+                                                      h5particle['creator_proc'].decode('utf-8','ignore'))
+
+        for h5hit in hit_rows:
+            ipart = h5hit['particle_indx']
+            current_particle = current_event[ipart]
+
+            hit = MCHit(h5hit['hit_position'],
+                        h5hit['hit_time'],
+                        h5hit['hit_energy'],
+                        h5hit['label'].decode('utf-8','ignore'))
+
+            current_particle.hits.append(hit)
+
+        evt_number             = h5extents[iext]['evt_number']
+        all_events[evt_number] = current_event
+
+    return all_events
+
+
+def compute_mchits_dict(mcevents:Mapping[int, Mapping[int, MCParticle]])->Mapping[int, MCHit]:
+    """Returns all hits in the event"""
+    mchits_dict = {}
+    for event_no, particle_dict in mcevents.items():
+        hits = []
+        for particle_no in particle_dict.keys():
+            particle = particle_dict[particle_no]
+            hits.extend(particle.hits)
+        mchits_dict[event_no] = hits
+    return mchits_dict
+
+
+def read_mcsns_response(h5f, event_range=(0, 1e9)) ->Mapping[int, Mapping[int, Waveform]]:
+
+    h5config = h5f.root.MC.configuration
+
+    bin_width_PMT  = 100. * units.nanosecond
+    bin_width_SiPM = 1. * units.microsecond
+    for row in h5config:
+        param_name = row['param_key'].decode('utf-8','ignore')
+        if param_name.find('time_binning') >= 0:
```

Comment:
If you only want to check containment use `if substring in string:`.

Reply:
I tried it at the beginning, but it didn't work properly.